app/components/add-cron-job.js

```diff
@@ -1,45 +1,68 @@
 import Component from '@ember/component';
 import { inject as service } from '@ember/service';
-import { task } from 'ember-concurrency';
+import { computed } from '@ember/object';
+import { mapBy } from '@ember/object/computed';
 
 export default Component.extend({
   store: service(),
+  branchesFetcher: service(),
 
   classNames: ['form--cron'],
 
+  intervals: ['monthly', 'weekly', 'daily'],
+
+  options: ['Always run', 'Do not run if there has been a build in the last 24h'],
+
+  currentCronJobsBranches: mapBy('repository.cronJobs', 'branch.name'),
+
+  init() {
+    this.reset();
+    this.setProperties({
+      'branchesFetcher.repository': this.get('repository')
+    });
+    this.get('branchesFetcher.showSearchResults').perform();
+    this._super(...arguments);
+  },
+
   reset() {
     this.setProperties({
       selectedBranch: null,
-      selectedInterval: null,
+      selectedInterval: this.intervals.firstObject,
+      selectedOption: this.options[0],
       disable: null
     });
   },
 
-  save: task(function* () {
-    const store = this.get('store');
-    const repoId = this.get('branches.firstObject.repoId');
-    const branch = this.get('selectedBranch') || this.get('branches.firstObject');
-
-    const existingCrons = yield store.filter('cron', { repository_id: repoId }, (c) => {
-      c.get('branch.repoId') === repoId && c.get('branch.name') === branch.get('name');
-    });
+  comparedBranches: computed('currentCronJobsBranches', 'branchesFetcher.searchResults', function () {
+    let branchNames = this.get('currentCronJobsBranches');
+    let branches = this.get('branchesFetcher.searchResults');
 
-    if (existingCrons.get('firstObject')) {
-      store.unloadRecord(existingCrons.get('firstObject'));
-    }
+    if (!branches) { return []; }
 
-    const cron = store.createRecord('cron', {
-      branch,
-      interval: this.get('selectedInterval') || 'monthly',
-      dont_run_if_recent_build_exists: this.get('selectedOption') || false
+    branches.forEach((branch) => {
+      branch.set('disabled', branchNames.indexOf(branch.name) > -1);
```

Comment:
Am I correct, that this `disabled` property is for `power-select` to disable options in the list that already have a cron? If so, then I have two issues:

1) Are you sure that there can't be multiple crons for the same branch? I don't see any reason why not. Why not build the same branch two times a day, let's say? Not sure about that, but sounds like a valid case.

2) I don't think it's a good way to do disabling. When you set a property on the Ember Model instance, it persists in the store and affect all other places where this model is used. For example, imagine that there is another dropdown with branches in other place. This will disable branches there too, because the property will reflect there as well. If (1) is not the case and the branches still need to be disabled, then the proper way to disable them would be to derive separate `branchOptions` list instead of mutating objects in this one. Something like:
```js
branchOptions: computed('currentCronJobsBranches.@each', 'branchesFetcher.searchResults.@each.name', function() {
  const { currentCronJobsBranches = [], branchesFetcher } = this;
  return branchesFetcher.searchResults.map(({ name }) => {
    const disabled = currentCronJobsBranches.includes(name);
    return { name, disabled };
  })
})
```
This will create a new list of objects with branch `name` and `disabled`; If any other branch properties are needed, they can be exposed in the same way.

You can find a pretty similar example here: https://github.com/travis-ci/travis-web/blob/ae27f51c001dfb17bd8719113c80a8dc1e693858/app/components/dialogs/migrate-beta.js

Reply:
[Response by @weban](https://github.com/travis-ci/travis-web/pull/2024#discussion_r287738199):
> About branch.set('disabled', branchNames.indexOf(branch.name) > -1);
Like I said from UX perspective better solution is show disabled option than nothing.
About map to {name, disabled} object there is no sense because it will require additional method to find branch object by name

@weban I totally agree with your point that showing disabled options is more useful. Although, my initial comment has slightly different questions, please re-read it. 

Basically, my question was, do they really need to be disabled? Because I don't see any reasons for that. And if they _do_ for some reason, then it **must** be done in a different way, because current implementation is prone to bugs and very fragile (refer to the example in the initial comment).